salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,200 @@
+                    missing_deps
+                ))
+            )
+
+    # move this to yaml file
+    @parameterized.expand([
+        (None, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {}, ('kubeconfig.conf', 'mycontext')),
+        (None, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        (None, {"kubeconfig": "kubeconfig.conf"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        (None, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, None, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, None, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}}}, {}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf", "kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"kubeconfig": "kubeconfig.conf", "context": "mycontext"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"context": "mycontext"}, {"kubernetes.context": "mycontext"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"kubeconfig": "kubeconfig.conf"}}}, {"context": "mycontext"}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
+        ({'metalk8s': {"api_server": {"context": "mycontext"}}}, {}, {"kubernetes.kubeconfig": "kubeconfig.conf"}, ('kubeconfig.conf', 'mycontext')),
```

Comment:
Right now it's really unreadable IMO

To make things easier to read please use different name depending on the place of the element 
E.g.: In the pillar "my-pillar-kubeconfig.conf" "my-pillar-context" in the kwargs "my-arg-kubeconfig.conf" ....

Reply:
Ok

Reply:
Agree with Tedd-E, and we can also add comments to explain what we're testing with each case, it eases to see if we're testing redundant things or if we missed a case.